tests/dataconverter/test_validation.py

```diff
@@ -21,46 +21,49 @@
-        "/my_entry/nxodd_name/float_value": 2.0,
-        "/my_entry/nxodd_name/float_value/@units": "nm",
-        "/my_entry/nxodd_name/bool_value": True,
-        "/my_entry/nxodd_name/bool_value/@units": "",
-        "/my_entry/nxodd_name/int_value": 2,
-        "/my_entry/nxodd_name/int_value/@units": "eV",
-        "/my_entry/nxodd_name/posint_value": np.array([1, 2, 3], dtype=np.int8),
-        "/my_entry/nxodd_name/posint_value/@units": "kg",
-        "/my_entry/nxodd_name/char_value": "just chars",
-        "/my_entry/nxodd_name/char_value/@units": "",
-        "/my_entry/nxodd_name/type": "2nd type",
-        "/my_entry/nxodd_name/date_value": "2022-01-22T12:14:12.05018+00:00",
-        "/my_entry/nxodd_name/date_value/@units": "",
-        "/my_entry/nxodd_two_name/bool_value": True,
-        "/my_entry/nxodd_two_name/bool_value/@units": "",
-        "/my_entry/nxodd_two_name/int_value": 2,
-        "/my_entry/nxodd_two_name/int_value/@units": "eV",
-        "/my_entry/nxodd_two_name/posint_value": np.array([1, 2, 3], dtype=np.int8),
-        "/my_entry/nxodd_two_name/posint_value/@units": "kg",
-        "/my_entry/nxodd_two_name/char_value": "just chars",
-        "/my_entry/nxodd_two_name/char_value/@units": "",
-        "/my_entry/nxodd_two_name/type": "2nd type",
-        "/my_entry/nxodd_two_name/date_value": "2022-01-22T12:14:12.05018+00:00",
-        "/my_entry/nxodd_two_name/date_value/@units": "",
-        "/my_entry/my_group/required_field": 1,
-        "/my_entry/definition": "NXtest",
-        "/my_entry/definition/@version": "2.4.6",
-        "/my_entry/program_name": "Testing program",
-        "/my_entry/my_group/optional_field": 1,
-        "/my_entry/required_group/description": "An example description",
-        "/my_entry/required_group2/description": "An example description",
-        "/my_entry/optional_parent/req_group_in_opt_group/data": 1,
+        "/ENTRY[my_entry]/optional_parent/required_child": 1,
+        "/ENTRY[my_entry]/optional_parent/optional_child": 1,
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/float_value": 2.0,
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/float_value/@units": "nm",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/bool_value": True,
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/bool_value/@units": "",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/int_value": 2,
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/int_value/@units": "eV",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/posint_value": np.array(
+            [1, 2, 3], dtype=np.int8
+        ),
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/posint_value/@units": "kg",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/char_value": "just chars",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/char_value/@units": "",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/type": "2nd type",
+        "/ENTRY[my_entry]/NXODD_name[nxodd_name]/date_value": "2022-01-22T12:14:12.05018+00:00",
```

Comment:
The key with `NXODD_name[nxodd_name]` should give an error and the error can be catch in test. Because, only accepted convention is `NXODD_NAME[nxodd_name]`, the given name should  be inside the square bracket. Or I am missing something.

Reply:
I don't understand your comment. They key `/ENTRY[my_entry]/NXODD_name[nxodd_name]/date_value` should be correct as any key which fits `NXODD_name` is (i.e.,`*_name`)

Reply:
I meant we do not write any `/ENTRY[my_entry]/NXODD_name[nxodd_name]/date_value` but rather `/ENTRY[my_entry]/NXODD_NAME[nxodd_name]/date_value`. If we modify or overwrite the name of the group that goes inside the square bracket. Or?

Reply:
Yes exactly, this notation means that we write a path `/my_entry/nxodd_name/date_value` to the file. The parts outside the brakets are just there to fit to a certain concept. It should also work with `/ENTRY[my_entry]/DATA[nxodd_name]/date_value` as `nxodd_name` will be an `NXdata` eventually.